soql-pack/src/main/scala/com.socrata.soql/SoQLPackDecoder.scala

```diff
@@ -79,6 +81,21 @@ object SoQLPackDecoder {
   def decodeBoolean(item: Any): Option[SoQLValue] =
     Some(SoQLBoolean(item.asInstanceOf[Boolean]))
 
+  def decodeUrl(item: Any): Option[SoQLValue] = try {
+    item match {
+      case Seq(url: Array[Byte], label: Array[Byte]) =>
+        Some(SoQLUrl(Some(new String(url, StandardCharsets.UTF_8)), Some(new String(label, StandardCharsets.UTF_8))))
+      case Seq(url: Array[Byte], null) =>
+        Some(SoQLUrl(Some(new String(url, StandardCharsets.UTF_8)), None))
+      case Seq(null, label: Array[Byte]) =>
+        Some(SoQLUrl(None, Some(new String(label, StandardCharsets.UTF_8))))
+      case Seq(null, null) =>
+        Some(SoQLUrl(None, None))
+    }
+  } catch {
+    case _: Exception => None
```

Comment:
i don't think this needs to be in a try-catch. The other one was because it's trying to decode to json. Also did you figure out the ues case for a Some(SoQUrl(none, none)) over just none?

Reply:
what I mean is: it seems like it would be better to just add a case statement that matches everything to make your pattern-match exhaustive. And ideally figure out if Some(SoQLUrl(None, None)) is needed.

Reply:
I walked through this a little more, to see that the decoder producing None makes a SoQLNull for the library user, which seems totally reasonable. 👍 so I went ahead with what you're suggesting!